src/Http/StringStream.php

```diff
@@ -0,0 +1,75 @@
+<?php
+namespace LORIS\Http;
+
+/**
+ * A StringStream provides a simple wrapper over a string to convert
+ * it to a \Psr\Http\Message\StreamInterface
+ */
+class StringStream implements \Psr\Http\Message\StreamInterface {
+    // @todo Handle error cases correct. PSR7 says to throw an exception,
+    // but the fopen/etc functions generally return the false literal.
```

Comment:
SPL File Handling objects throw RuntimeException that would be conform to PSR-7.

Reply:
How do you know? The documentation that I can find (https://secure.php.net/manual/en/class.splfileobject.php) is... sparse

Reply:
I agree, it is hard to tell. There is some entry in the doc and I tried an other.

Reply:
The file related functions explicitly say that they return false on error, and it was easy enough to just explicitly check the return value, so I did that rather than rewrite around SPL.